### PR TITLE
Revert horovod example to v1

### PR DESCRIPTION
**Pull request by alculquicondor**
The controller generates keys and mounts them to the containers. The container images must know how to place the credentials and set file permissions.

First part of #373 (Replace kubectl and support roles with ssh and secrets)

Tested with the horovod CPU image

### CODE REVIEW

File: manifests/overlays/dev/kustomization.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ commonLabels:
   app.kubernetes.io/component: mpijob
 images:
 - name: mpioperator/mpi-operator
-  newName: gcr.io/acondor-gke-dev/mpi-operator
+  newName: dev-registry/mpi-operator
```
Comment (terrytangyuan) [inline manifests/overlays/dev/kustomization.yaml:14]: Is this change intentional? Does this exist?

Comment (alculquicondor) [reply manifests/overlays/dev/kustomization.yaml:14]: Neither exist (my GCR is private). It's intended for developers to use their preferred registry.

**Review by terrytangyuan — APPROVED** [review]
LGTM. Others PTAL

File: examples/horovod/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (gaocegege) [inline examples/horovod/entrypoint.sh:1]: I am wondering if it is needed for every MPIJob. Should we add some comments about why to introduce such a entrypoint script?

Comment (alculquicondor) [reply examples/horovod/entrypoint.sh:1]: Yes, something like this will always be needed.
The problem is that SSH is very picky about not just file permissions, but also the permissions of the directory. Unfortunately, kubernetes doesn't allow to change the permissions of the mount folder.

I'll update the README once I'm done with all development (tests included). The idea is: the controller will mount the keys in /mnt/ssh. It's up to the container image to set those files in the appropriate location for their user.

Comment (alculquicondor) [reply examples/horovod/entrypoint.sh:1]: FTR, I tried to use mount the files in /root/.ssh or used them from /mnt/ssh by setting `AuthorizedKeysFile` and `IdentityFile` in the ssh configuration. However, SSH complained about the folder permissions :(

Comment (alculquicondor) [reply examples/horovod/entrypoint.sh:1]: Hold on, I'm investigating if I can do the same in an init container.

Comment (alculquicondor) [reply examples/horovod/entrypoint.sh]: I'm surprised, but it worked. Now, I'm not convinced that it will work if the containers don't run as root, but I'll check that later. I added a bullet point in #373.

**Comment by alculquicondor** [comment]
Btw, I found a bug where the incorrect scheme was being used. As a consequence, defaults were not applied and events fail to record. See latest commit.

**Comment by alculquicondor** [comment]
@carmark @Jeffwan any other comments?

File: examples/horovod/tensorflow-mnist.yaml [hunk 3]
```diff
@@ -15,7 +15,6 @@ spec:
             name: mpi-launcher
             command:
             - mpirun
-            args:
```
Comment (gaocegege) [inline examples/horovod/tensorflow-mnist.yaml:18]: Do we need to remove it?

Comment (alculquicondor) [reply examples/horovod/tensorflow-mnist.yaml]: No, I just messed it up when reverting. Also, using a the v1 api works just fine. the apiserver does the conversion for us.

**Review by gaocegege — COMMENTED** [review]
PTAL

This PR removes the CLI argument `kubectl-delivery-image`, does it work for you? @carmark @terrytangyuan

**Review by gaocegege — COMMENTED** [review]
LGTM

Thanks for your contribution! :tada: :+1:

**Review by terrytangyuan — COMMENTED** [review]
LGTM. @zw0610 take another look?

**Comment by alculquicondor** [comment]
uhm.... #375 was already merged, which contain these changes... should we revert? @terrytangyuan

**Comment by terrytangyuan** [comment]
Then we can close this one?

**Comment by alculquicondor** [comment]
I wasn't sure if there were any changes here that I didn't put in the other branch. But it looks like it was all there.

**Comment by alculquicondor** [comment]
This commit was missing

**Comment by terrytangyuan** [comment]
Looks like there’s only one changed file?

**Comment by alculquicondor** [comment]
correct

**Comment by terrytangyuan** [comment]
Could you update the PR title then?

**Comment by alculquicondor** [comment]
Is there a way to retrigger the test? It looks like the event watcher is not working or something. I can't reproduce locally.

**Comment by terrytangyuan** [comment]
Just re-triggered it
